internal/host/validator.go

```diff
@@ -1229,18 +1229,179 @@ func (v *validator) printDefaultRoute(c *validationContext, status ValidationSta
 	}
 }
 
-func shouldValidateDnsResolution(c *validationContext) bool {
-	// Skip DNS resolution checks in IPI network mode
-	if !swag.BoolValue(c.cluster.UserManagedNetworking) {
+func ignitionHasFile(ignition *ignition_types.Config, path string) bool {
+	for _, file := range ignition.Storage.Files {
+		if file.Path == path {
+			return true
+		}
+	}
+
+	return false
+}
+
+func ignitionContainsManagedNetworkingFiles(config *ignition_types.Config) bool {
+	if ignitionHasFile(config, "/etc/kubernetes/manifests/coredns.yaml") ||
```

Comment:
if we check both files then we need && .
I still think one of them is enough but if we decide to have both lets validate that both exists

Reply:
See comment of why I check either and not both:

> To be a bit morerobust, we consider the presence of any one of them to mean that the cluster has managed networking. This gives us better forwards compatibility if either one of them gets renamed / replaced with other technologies in future OCP versions.